packages/frontend/src/state/lp/hooks.ts

```diff
@@ -0,0 +1,147 @@
+
+/*** GETTERS ***/
+
+export const useGetPosition = () => {
+  const contract = useAtomValue(nftManagerContractAtom)
+
+  const getPosition = useCallback(
+    async (uniTokenId: number) => {
+      if (!contract) return null
+      const position = await contract.methods.positions(uniTokenId).call()
+      const { nonce, operator, token0, token1, fee, tickLower, tickUpper, liquidity, feeGrowthInside0LastX128, feeGrowthInside1LastX128, tokensOwed0, tokensOwed1 } = position
+      return {
+        nonce,
+        operator,
+        token0,
+        token1,
+        fee,
+        tickLower,
+        tickUpper,
+        liquidity,
+        feeGrowthInside0LastX128,
+        feeGrowthInside1LastX128,
+        tokensOwed0,
+        tokensOwed1
+      }
+    },
+    [contract],
+  )
+
+  return getPosition
+}
+
+export const useGetDecreaseLiquidity = () => {
+  const contract = useAtomValue(nftManagerContractAtom)
+
+  const getDecreaseLiquiduity = useCallback(
+    async (tokenId: number, liquidity: number, amount0Min: number, amount1Min: number, deadline: number) => {
+      if (!contract) return null
+      const DecreaseLiquidityParams = {
+        tokenId,
+        liquidity,
+        amount0Min,
+        amount1Min,
+        deadline,
+      }
+
+      const decreaseLiquidity = await contract.methods.decreaseLiquidity(DecreaseLiquidityParams).call()
+
+      return decreaseLiquidity
+    },
+    [contract],
+  )
+
+  return getDecreaseLiquiduity
+}
+
+export const useGetExactIn = () => {
+  const contract = useAtomValue(quoterContractAtom)
+  const {weth, oSqueeth} = useAtomValue(addressesAtom)
+
+  const getExactIn = useCallback(
+    async (amount: BigNumber, squeethIn: boolean) => {
+      if (!contract) return null
+
+      const QuoteExactInputSingleParams = {
+        tokenIn: squeethIn ? oSqueeth : weth,
+        tokenOut: squeethIn ? weth : oSqueeth,
+        amountIn: amount.toFixed(0),
+        fee: POOL_FEE,
+        sqrtPriceLimitX96: 0
+      }
+
+      const quote = await contract.methods.quoteExactInputSingle(QuoteExactInputSingleParams).call()
+      return quote.amountOut
+    },
+    [contract, weth, oSqueeth],
+  )
+
+  return getExactIn
+}
+
+export const useGetExactOut = () => {
+  const contract = useAtomValue(quoterContractAtom)
+  const {weth, oSqueeth} = useAtomValue(addressesAtom)
+
+  const getExactOut = useCallback(
+    async (amount: BigNumber, squeethOut: boolean) => {
+      if (!contract) return null
+
+      const QuoteExactOutputSingleParams = {
+        tokenIn: squeethOut ? weth : oSqueeth,
+        tokenOut: squeethOut ? oSqueeth : weth,
+        amount: amount.toFixed(0),
+        fee: POOL_FEE,
+        sqrtPriceLimitX96: 0
+      }
+
+      const quote = await contract.methods.quoteExactOutputSingle(QuoteExactOutputSingleParams).call()
+      return quote.amountIn
+    },
+    [contract, weth, oSqueeth],
+  )
+
+  return getExactOut
+}
+
+async function getPoolState(poolContract: Contract) {
```

Comment:
Same comment. Seems like duplicate

Reply:
Got it - removing this one and adapting old one to include tickSpacing.